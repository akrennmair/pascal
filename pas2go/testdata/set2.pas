program test;

var s1, s2, s3 : set of integer;

begin
	s1 := [ 10, 5, 17 ];
	s2 := [ 5, 18, 20 ];
	s3 := s1 + s2;

	if not (20 in s3) then
		writeln('error: 20 not found in union!');

	if not (17 in s3) then
		writeln('error: 17 not found in union!');

	s3 := s1 - s2;

	if not (10 in s3) then 
		writeln('error: 10 not found in difference!');

	if 5 in s3 then 
		writeln('error: 5 found in difference!');
	
	s3 := s1 * s2;

	if not (5 in s3) then
		writeln('error: 5 not found in intersection!');

	if 18 in s3 then
		writeln('error: 18 found in intersection!')

end.
