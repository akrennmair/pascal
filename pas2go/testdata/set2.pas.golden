package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		s1 system.SetType[int]
		s2 system.SetType[int]
		s3 system.SetType[int]
	)
	_ = s1
	_ = s2
	_ = s3

	system.SetAssign(&s1, system.Set[int](10, 5, 17))
	system.SetAssign(&s2, system.Set[int](5, 18, 20))
	system.SetAssign(&s3, s1.Union(s2))
	if !(s3.In(20)) {
		system.Writeln("error: 20 not found in union!")
	}
	if !(s3.In(17)) {
		system.Writeln("error: 17 not found in union!")
	}
	system.SetAssign(&s3, s1.Difference(s2))
	if !(s3.In(10)) {
		system.Writeln("error: 10 not found in difference!")
	}
	if s3.In(5) {
		system.Writeln("error: 5 found in difference!")
	}
	system.SetAssign(&s3, s1.Intersection(s2))
	if !(s3.In(5)) {
		system.Writeln("error: 5 not found in intersection!")
	}
	if s3.In(18) {
		system.Writeln("error: 18 found in intersection!")
	}
}
